tasks/broad/Alignment.wdl

```diff
@@ -60,13 +59,34 @@ task SamToFastqAndBwaMemAndMba {
         exit 1;
     fi
 
+    # "natually" sorted reads are not queryname sorted according to picard/htsjdk
+    # for the old version of picard that's here, BMA chokes with strange error messages
+    # this pre-emptiv sort should fix the problem.
+
+    java \
+      -Dsamjdk.use_async_io_read_samtools=true \
+      -Dsamjdk.use_async_io_write_samtools=true \
+      -Xmx13g \
+      -jar /usr/gitc/picard.jar \
+      SortSam \
+      --COMPRESSION_LEVEL 0 \
+      --VALIDATION_STRINGENCY SILENT \
+      -I ~{input_bam} \
+      -O  /dev/stdout \
+      --MAX_RECORDS_IN_RAM 5000000 \
+      -SO queryname | \
+     samtools view -b -1 --threads 8 - > ~{output_bam_basename}.qname_sorted.bam
```

Comment:
Is the version of samtools used by this process recent enough to have `samtools sort -N` (note the CAPITAL N):

```
  -N         Sort by read name (ASCII): cannot be used with samtools index
```